bin/generate-post-meta-image

```diff
@@ -1,34 +1,68 @@
 #!/bin/sh -e
 
-text="$1"
-bg="$2"
-out=$3
+tmp=$(mktemp -d)
+bg="#2421AB"
 res=2160x1080
 
-tmp=$(mktemp -d)
+while [[ $# -gt 0 ]] && [[ "$1" == "--"* ]]; do
+  case "$1" in
+    "--resolution" | "-r")
+      res=$2
+      shift
+      shift
+      ;;
 
-if [[ "$bg" == "" ]]; then
-  bg="#2421AB"
-  convert -size $res xc:$bg $tmp/bg.png
-else
-  convert $2 -resize $res^ $tmp/bg-uncropped.png
-  convert $tmp/bg-uncropped.png -gravity center -crop $res+0+0 -fill black -colorize 40% $tmp/bg.png
-fi
+    "--text" | "-t")
+      text=$2
+      shift
+      shift
+      ;;
 
-convert -size $res xc:"#00000000" \
-  -pointsize 100 -fill white -draw 'text 100,160 "Blog Post" ' \
-  $tmp/text-top.png
+    "--bg" | "-b")
+      bg=$2
+      shift
+      shift
+      ;;
+  esac
+done
 
-convert -size $res xc:"#00000000" \
-  -font ./src/fonts/acta-headline-extra-bold.woff \
-  -pointsize 165 -fill white -draw "text 100,925 '$text' " \
-  $tmp/text-bottom.png
+out=$1
 
+
+if [[ -f "$bg" ]]; then
+  # bg is a file. scale & crop the image to intended dimensions,\
+  # and add a transparent layer on top
+  convert $bg \
+    -resize $res^ \
+    -gravity center \
+    -crop $res+0+0 \
+    -fill black \
+    -colorize 40% \
+    $tmp/bg.png
+else
+  # bg will be assumed to be a color, used as is
+  convert -size $res \
+    xc:$bg \
+    $tmp/bg.png
+fi
+
+# join the 3 elements:
+#   - generated background;
+#   'Blog Post' on top
+#   post title on the bottom
 convert \
   -page +0+0 $tmp/bg.png \
-  -page +0+0 static/script-assets/blog-post-top.png \
-  -page +0+0 $tmp/text-bottom.png \
-  -layers merge +repage $out
+  -size $res \
+  xc:"#00000000" \
+  -fill white \
+  -pointsize 95 \
+  -gravity northwest \
+  -annotate +100+160 'Blog Post' \
+  -gravity southwest \
+  -font ./src/fonts/acta-headline-extra-bold.woff \
+  -annotate +100+100 "$text" \
+  -layers merge \
+  +repage \
+  $out
 
-# Optimize image for smaller footprint
```

Comment:
Why remove this comment? Seems to be consistent with the rest of the script, where there's a comment explaining what's intended with each "section".

Reply:
because to be honest, the command itself is pretty much self-documenting

Reply:
Not sure about it if someone doesn't know what Bundler or `image_optim` are, but if you feel it is I'm ok with removing.